### ISSUE TITLE
Licence of the extensions

### ISSUE THREAD

**Issue by Natrium729** [i7/extensions#62]
The repo has no licence information, and I think that it's important to know under what licence the extensions are released.

Since the extensions available in the Public Library are under the CC-BY licence, I guess that's the licence of a lot of the extensions in the repo.

However, since there are many authors, I understand it's difficult to take care of all this. Maybe authors can add a LICENSE file in their folder if  they don't wan't to have the default licence of the repo (which may or may not be CC-BY).

(I'm far from being an expert on legal matters, so it may be even more complicated that I think it is.)

**Comment by uecasm** [comment]
The licensing status of all Inform extensions from the Public Library is given as [CC-BY](http://inform7.com/book/WI_27_1.html).

While this repository isn't curated to the extent that used to be (and has a less formal entry process), as far as I am aware the spirit of that license still lives on in all extensions here.  (If you can find any statement to the contrary in any specific extension, please point it out.)

I don't think it's a good idea to invite individual extension authors to use a different license.

**Comment by curiousdannii** [comment]
We should definitely include a license file in the root. I think I agree that it's probably best to not have extensions under other licenses - if someone doesn't want theirs to be CC-BY licensed they should host it somewhere else I think.

**Comment by uecasm** [comment]
I don't think GitHub directly recognises CC-BY licenses like it does some of the others (to show the license name directly) since it has software-only blinders on.  But this will add a link to read it, at least.

**Comment by dscorbett** [comment]
Note that simply adding a license file to a repository doesn’t license everything in the repository under that license. You actually have to check each extension, ask its copyright holder for permission to use CC-BY if applicable, and remove it if you don’t get permission.

**Comment by Natrium729** [comment]
I agree you technically can't just add a licence file like that. For what is worth, we only know for sure that only the extensions from the Public Library are under CC-BY (even if CC-BY have always be the "spirit" of the repo).

It may also be a good idea to add a line somewhere (e.g. in the README) that indicates that if you want to add you extensions to the repo, you agree to release them under the CC-BY licence (just to be clear to everyone).

And since this repo is for the moment the de facto place to look for Inform extensions, maybe there could be a section in the README listing other places where authors can find extensions, for people that don't want to host them here (for example if they don't want to release them under CC-BY).

**Comment by neroden** [comment]
FWIW, my extensions in this repository are all under CC-BY.  I hereby declare so.

Some of it is based on other people's work, and I'm not sure which of those works were in the Public Library or explicitly CC-BY licensed, though.

Specifically, Neutral Standard Responses contains material from
Neutral Library Messages by Aaron Reed
Unknown Word Error by Mike Ciel, which itself contains material from Andrew Plotkin and from Neil Cerutti's Dunno for i6

My version of Large Game Speedup is based on Andrew Plotkin's (with extensive changes).

My version of Undo Output Control contains material from
Undo Output Control by Erik Temple
Empty Command Handling by Daniel Stelzer based on code by Matt Weiner

I would appreciate it if these could be "license-cleared".

A good first step would be to make a list of which of the extensions in this repo were in the Public Library (and therefore definitely under CC-BY).  Then we can try asking the authors of other extensions one at a time.

**Comment by erkyrath** [comment]
My stuff is all CC-BY also. I just added a note to https://github.com/erkyrath/i7-exts saying so.

**Comment by neroden** [comment]
I think most of the extension authors have email addresses which can be tracked down.  Would it be out of line to start contacting the authors of the older extensions which weren't in the Public Library and asking for explicit permission to license under CC-BY?
